adding dotenv files

memsharded opened the pull request:
Changelog: Feature: Add generation of ``dotenv`` environment files with ``tools.env:dotenv=True`` conf.
Docs: https://github.com/conan-io/docs/pull/4246

Close https://github.com/conan-io/conan/issues/11042

memsharded committed:
new test for lockfile requires error

perseoGI committed:
WIP: added premake toolchain and updated premake generator

perseoGI committed:
Updated premake toolchain

perseoGI committed:
Created build method on Premake, fix cross building and map architectures

perseoGI committed:
Added extra flags on premake generator

perseoGI committed:
Stable version, not complete

perseoGI committed:
Detailed tooling

perseoGI committed:
Improved code generation identation

perseoGI committed:
Add option to disable projects

perseoGI committed:
Removed global projects as it is not an stable solution

perseoGI committed:
Use ConanException instead

perseoGI committed:
Removed possibility of managing different workspaces

perseoGI committed:
Restore multi workspace allowing PremakeToolchain to be used as a plain generator

memsharded committed:
new test for build-scripts transitive

memsharded committed:
adding dotenv files

AbrilRBS committed:
Sort overrides when serializing

memsharded committed:
fix test in multiplatform

memsharded commented on conan/internal/model/conf.py:
```diff
@@ -126,6 +126,7 @@
     "tools.apple:enable_arc": "(boolean) Enable/Disable ARC Apple Clang flags",
     "tools.apple:enable_visibility": "(boolean) Enable/Disable Visibility Apple Clang flags",
     "tools.env.virtualenv:powershell": "If specified, it generates PowerShell launchers (.ps1). Use this configuration setting the PowerShell executable you want to use (e.g., 'powershell.exe' or 'pwsh'). Setting it to True or False is deprecated as of Conan 2.11.0.",
+    "tools.env:dotenv": "Generate dotenv environment files",
```
Should we make it an opt-in conf?
Or maybe generate it by default, always to make it simpler/more convenient?

AbrilRBS replied:
I like the idea that it would be nice to always generate this file with and opt-out, (but would we even want an specific opt-out?)

memsharded replied:
We might wait for the opt-out until users give feedback.

memsharded committed:
Coverage (#18275)

* testing RequirementsInfo

* removed conan config-install gz, unused

* simplify gzopen_without_timestamps

memsharded committed:
trying again to define CMAKE_XX_COMPILER in generated presets, only for cl

AbrilRBS committed:
Allow composition of conf values that are different categories of numbers (#18265)

* Add profile composition test for numbers

* Add failing test

* Allow composition over conf values that are different categories of numbers

AbrilRBS commented on conan/tools/env/environment.py:
```diff
@@ -568,6 +580,9 @@ def save_script(self, filename):
         else:
             self.save_sh(path)
 
+        if self._conanfile.conf.get("tools.env:dotenv", check_type=bool):
+            self.save_dotenv(f"{name}.env")
```
```suggestion
        self.save_dotenv(f"{name}.env")
```

and remove the conf for now

memsharded replied:
Lets try and see if some test break...

memsharded committed:
Update conan/tools/env/environment.py

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

perseoGI committed:
Added more premake tests

perseoGI committed:
Parametrize premake test

perseoGI committed:
Added some comments

perseoGI committed:
Removed old premake functional test

perseoGI committed:
Added docstring and a way of controlling verbosity on premake

perseoGI committed:
Fixed tests

AbrilRBS committed:
Add clearer message when missing some keys in graph file

AbrilRBS committed:
simplify test

perseoGI committed:
Revert extra formatting

perseoGI committed:
Remove print

memsharded committed:
new tests for options and settings negation (#18306)

shtanko-sv committed:
Adding C standard detection (#18290)

* Add default C standard detection for various compilers

* Add detection for C standard in default settings

* Update comments

* Remove cstd from default generated profiles

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

* Add some unit tests

* Remove detect_cstd function

* Update detect.py module after removing `detect_cstd`

* Update conan/internal/api/detect/detect_api.py

* Cleanups

---------

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

perseoGI committed:
Refactor cache save to create a list of files to compress and reuse existing function

perseoGI committed:
Applied suggestions

AbrilRBS committed:
Add gcc 14.3

uilianries committed:
[fix] tools.files.copy should exclude symlinks for folders (#18304)

* Validate copy exclude for symlinks

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Validate copy exclude for file symlinks

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Validate exclude folder as pattern

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Exclude symlink when matching the pattern

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Keep pep8 format

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>

memsharded committed:
Test/modernize (#18292)

* modernize tests

* speed up CMake checks

* fixing tests

* tests short summary

* tests short summary

* fix test

franramirez688 committed:
Avoid duplicate component requirements

perseoGI committed:
Add test for checking pkglist.json destination compression

perseoGI committed:
Remove pkglist file after compression from tmp folder

perseoGI committed:
Update conan/internal/api/uploader.py

Co-authored-by: James <memsharded@gmail.com>

danimtb committed:
[fix] Change wording on unzip tool when uncompressing file

danimtb committed:
use existing test

AbrilRBS committed:
Expose default_cstd for the build method

AbrilRBS committed:
Add missing default import

memsharded committed:
fixing grafted commits (#18315)

* fixing grafted commits

* --decorate for grafted log msg

* check

* avoid first --depth=1

* adding --refetch

franramirez688 committed:
wip

franramirez688 committed:
wip

franramirez688 committed:
wip

franramirez688 committed:
wip

franramirez688 committed:
wip

AbrilRBS committed:
Add super() test for set_version

franramirez688 committed:
wip

franramirez688 committed:
wip

franramirez688 committed:
wip

perseoGI committed:
Keep both legacy premake generator logic

While introducing new generator by simply enabling PremakeToolchain

perseoGI committed:
Added new tests for premake generators

memsharded committed:
fixing clang-cl autotools

franramirez688 committed:
[workspace] Removed `home_folder` mechanism (#18339)

Removed home_folder definition from WS

franramirez688 committed:
Putting conanws as top root for WS if exists

memsharded committed:
also work for clang-cl

uilianries committed:
Add support for extra configure arguments (#18333)

* Add support for extra configure arguments

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Check configuration type

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Validate type-check for list-like

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Replace extra_configure_args by configure_args

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Do not check type

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

* Do not check type

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

* Revert "Replace extra_configure_args by configure_args"

This reverts commit 91ac324e73d1f159270a2466e01b7175370d1843.

* Simplify tests^

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>
Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

memsharded committed:
wip

memsharded committed:
fixing cppstd flag

memsharded committed:
modernizing tests (#18349)

* modernizing tests

* more tests

* moved test

memsharded committed:
wip

memsharded committed:
wip

memsharded committed:
tests passing

memsharded committed:
fix conanws folder (#18352)

ErniGH committed:
Fix | Validate if the licenses in the SBOM are SPDX compatible (#18358)

* validate spdx_licenses in the sbom, test

* add deprecated licenses and fixes

franramirez688 committed:
Renamed 'editables' to 'packages'

memsharded committed:
new test for editable with build-host folders

memsharded committed:
moved test to integration

czoido committed:
fix parse errors

czoido committed:
Fix urls for audit command (#18360)

* fix url

* fix more urls

timviola-cmi committed:
Fix Conan Source folders.root issue

- Fixed issue in local.py where source() method did not account for
  self.folders.root (when specified)

- Added
test/integration/command/source_test.py::SourceTest::test_local_source_layout_root

Close #18376

perseoGI committed:
Fix codecov.yml invalid format (#18372)

AbrilRBS committed:
Add gcc 13.4 (#18374)

* Add gcc 13.4

* Update conan/internal/default_settings.py

memsharded committed:
fixing autotools_exe template, avoiding libtool (#18378)

memsharded committed:
fixing CmakeConfigDeps link flags (#18367)

perseoGI committed:
Drop support for buildenv flags reading

czoido committed:
Conan 2.18.0-dev

czoido committed:
Fix PyInstaller --exclude-module (#18381)

add wildcard

rudenkornk committed:
fix(tests): fix some non-functional asserts in conflict_diamond_test.py (#18394)

uilianries committed:
[fix] Honor when raising ConanInvalidConfiguration in hooks (#18385)

* Honor ConanInvalidConfiguration in hooks

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Cover post_validate

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>

memsharded committed:
Feature/profile contexts (#18383)

* investigating if possible to add context to profile rendering

* added test

* fix tests

* fix tests

AbrilRBS committed:
Add open functionality to TestClient

AbrilRBS committed:
Comment

perseoGI committed:
Added wasm64 architecture

AbrilRBS committed:
Assert out when running on CI and calling open

perseoGI committed:
Added MEMORY64=1 on wasm64 to architecture_flag function

timviola-cmi committed:
Update Integration Test

- Test should now pass on windows, linux, and mac OS
- Test creates a file in `source()` method and loads the file
  in the `build()` method. The test will fail if both `source()`
  and `build()` do not use the same `self.source_folder`

memsharded committed:
Update test/integration/command/source_test.py

perseoGI committed:
Support asm.js arch correctly

AbrilRBS committed:
Allow multi version minors for detected compilers

memsharded committed:
[CMakeConfigDeps] Fixed `find_package` in subforlders (#18407)

* new test for CMakeConfigDeps with subfolders

* wip

* fix tests

* fix test

* fix test

AbrilRBS committed:
Dont pick up spaces

AbrilRBS committed:
Dont pick up spaces

AbrilRBS committed:
Make older gccs explictily return up to minor, to keep compat with old behaviour

perseoGI committed:
Added emcc compiler support

memsharded committed:
fix in autotools win-clang and test for NMake with clang-cl

memsharded committed:
fix tests

memsharded committed:
Update test/functional/utils.py

perseoGI committed:
Added emcc compiler support

ErniGH committed:
[fix] local-recipe-index find bug, select pattern with only name (#18423)

* fix with test

* Update list_test.py

* simplify test

* more test cases

franramirez688 committed:
[BazelDeps] Pure refactor (#18424)

* Refactoring. WIP

* wip

* Fixed

memsharded committed:
better remote caching (#18411)

AbrilRBS committed:
Ensure performance test have known cache location that works everywhere (#18428)

* Make performance cache live inside the test folder

* Add gitignore for performance cache

AbrilRBS committed:
Create folders if they don't exist when using `--out-file` (#18427)

mkdir for --out-file

Co-authored-by: memsharded <james@conan.io>

memsharded committed:
fixing source retrieval when resetting local-index remote (#18418)

danimtb committed:
feature: Add support for source targets CMakeConfigDeps generator (#18350)

* first implementation

* implement compontents

* fix

* clean tests

* only linux for tests and cleanup

* move tests

* move join to jinja template

* Update conan/tools/cmake/cmakedeps2/target_configuration.py

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

* Revert "Update conan/tools/cmake/cmakedeps2/target_configuration.py"

This reverts commit a55b0c97e264b78a32d4ebaf8ed9fa4479b30987.

* headers not needed

* add deduce test

* improve test

* integrate sources into lib code

---------

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

czoido committed:
Less verbose text upload (#18430)

less verbose text upload

perseoGI committed:
WIP Added functional tests

perseoGI committed:
WIP: Added emscripten on windows

perseoGI committed:
WIP: Added emscripten on windows

perseoGI committed:
WIP: Added emscripten on windows

uilianries committed:
[feature] Add generic config for CMake and Meson install strip (#18429)

* Use tools.install:strip for CMake

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Add meson install strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Validate strip flag when installing with meson

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Improve strip config description

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Fallback for tools.cmake:install_strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Use deprecation flag for config

Co-authored-by: James <memsharded@gmail.com>

* Reduce install strip property

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Rename config to tools.build.install:strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Rename config to tools.build.install_strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Fix typo tools.build.install_strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>
Co-authored-by: James <memsharded@gmail.com>

perseoGI committed:
WIP: Added emscripten on windows

perseoGI committed:
WIP: Added emscripten on windows

perseoGI committed:
WIP: Added emscripten on windows

perseoGI committed:
Fix dockerfile

perseoGI committed:
Fix dockerfile

AbrilRBS committed:
Add support for sbom and lockfiles for conan audit list (#18437)

* Add support for sbom and lockfiles for conan audit list

* Ensure sbom is cyclone

* Fix remote for list

* Add test for pkglist with remote

---------

Co-authored-by: Carlos Zoido <mrgalleta@gmail.com>

memsharded committed:
fixing CMake presets on Windows with backslash (#18435)

* fixing CMake presets on Windows with backslash

* adding checks to test

perseoGI committed:
Fix dockerfile

perseoGI committed:
Fix dockerfile

lucaskdc committed:
Fix: parse -cc args when running "conan cache backup-upload"

perseoGI committed:
Add EM_CACHE in docker container

perseoGI committed:
Pre build emscripten binaries on image and remove windows specific tests

uilianries committed:
[fix] Add support for Git tree-less repository when getting the URL (#18444)

* Add test to validate Git treeless clone support

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Remove git method from the remote url

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* rsplit FTW

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Always return string

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Filter out (fetch) and (push)

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>

perseoGI committed:
Pre build emscripten binaries on image and remove windows specific tests

perseoGI committed:
Pre build emscripten binaries on image and remove windows specific tests

memsharded committed:
fix issue with missing folder in local-recipes-index (#18449)

perseoGI committed:
Only run node wasm64 on macos

perseoGI committed:
Relax tests to be os agnostic

AbrilRBS committed:
Add test to verify that we always call parse_args for every command

AbrilRBS committed:
Work for workspace too

perseoGI committed:
Removed unneded platform_tool_requires

AbrilRBS committed:
`conan export-pkg` now correctly passes a `str` as the conanfile version (#18456)

* conan export-pkg now correctly passes a str as the conanfile version

* Add type test

memsharded committed:
do not call generators for editables not being built (#18402)

perseoGI committed:
Skip emcc tests on python < 3.8 (#18461)

perseoGI committed:
Removed -rf pytest option and leave default (-rfE) which prints failed tests at the end

memsharded committed:
refactoring graph node.dependencies->node.edges (#18463)

* refactoring graph node.dependencies->node.edges

* fix sbom

memsharded committed:
adding a deprecation warning and fallback for Node.dependencies->edges

ErniGH committed:
[new command] `conan report diff` to inspect diffs between versions and revisions  (#18247)

* wip

* wip

Co-Authored-By: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

* Fix sidebar overflow behaviour, escape lines of html charcaters

* Update conan/cli/commands/compare.py

Co-authored-by: James <memsharded@gmail.com>

* Update conan/cli/commands/compare.py

Co-authored-by: James <memsharded@gmail.com>

* Cleanup html diff generation, follow formatter style from Conan commands

* Fix imports

* Simplify parameter handling, fix ref report for download

* Add search bar

* Fix list visuals after search

* Improve performance for large diffs

* Add json formatter

---------

Co-authored-by: Ernesto de Gracia Herranz <ernestodgh@jfrog.com>

* Typo

* Avoid encoding issues by ignoring them

The implementation right now is a bit ugly, will simplify when it's
ready

---------

Co-authored-by: Ernesto de Gracia Herranz <ernestodgh@jfrog.com>

* Add some basic tests

---------

Co-authored-by: Ernesto de Gracia Herranz <ernestodgh@jfrog.com>

* improve test and use "--- a" instead of "diff --git" (path with spaces)

* fix

* fix path

* change html test

* wip

* Make test fail when there are patches

* Test confgiure_source

* Test for patches

* Html test

* Tests

* test contents

* replace("\\", "/") for windows and use buffer to reorderlines

* fix

* fix

* fix new files, and fix tests

* Rename

* Break everything

* Don't show repeated files

* Simplify generation, share json/html inut data

* Fix styling

* More expressive check

* Move to api? This might get reverted later if we find a better place

* Improve output

* Add exclude search

* Don't take into account paths when filtering

* Remove duplication, don't try tro download cache refs

* Typo

* fix test

* Dont modify load function

* Rename function

* Use conan_run

* Hide git diff stderr

* Revert check_output_runner

* Fix issues with paths

---------

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>
Co-authored-by: Abril Rincón Blanco <abrilrb@jfrog.com>
Co-authored-by: James <memsharded@gmail.com>

AbrilRBS committed:
Fix head method not working in testing mocked requests

kambala-decapitator committed:
XcodeToolchain: fix deployment target for non-macOS (#18471)

* XcodeToolchain: fix deployment target for non-macOS

* remove duplicate line

* add new test

* Simplify testing

* Simplify testing

---------

Co-authored-by: Abril Rincón Blanco <abrilrb@jfrog.com>

AbrilRBS committed:
[PkgConfigDeps] Mechanism to skip some `*.pc` files creation (#18439)

* Add pkg_config_skip property to PkgConfigDeps

* Add pkg_config_skip property to PkgConfigDeps test

* Remove debug message

* Remove unsued import

* Move to use none pkg_config_name

* Update conan/tools/gnu/pkgconfigdeps.py

* Component whose main is skipped should define the pkg_config_name

* None components is not hidden

* Review

* Update test/integration/toolchains/gnu/test_pkgconfigdeps.py

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

---------

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

memsharded committed:
Feature: workspace create (#18390)

* workspace Python files proposal

* fixes

* review

* workspace create

* wip

* wip

* tests passing

* wip

* wip

* wip

* wip

* wip

* tests passing, first proposal for orchestrated build + create

* dynamic ref attributes for conan workspace create

* test for conan workspace build too

* wip

* Update conan/cli/commands/workspace.py

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

* wip

* [feature] Add generic config for CMake and Meson install strip (#18429)

* Use tools.install:strip for CMake

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Add meson install strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Validate strip flag when installing with meson

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Improve strip config description

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Fallback for tools.cmake:install_strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Use deprecation flag for config

Co-authored-by: James <memsharded@gmail.com>

* Reduce install strip property

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Rename config to tools.build.install:strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Rename config to tools.build.install_strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Fix typo tools.build.install_strip

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>
Co-authored-by: James <memsharded@gmail.com>

* Add support for sbom and lockfiles for conan audit list (#18437)

* Add support for sbom and lockfiles for conan audit list

* Ensure sbom is cyclone

* Fix remote for list

* Add test for pkglist with remote

---------

Co-authored-by: Carlos Zoido <mrgalleta@gmail.com>

* fixing CMake presets on Windows with backslash (#18435)

* fixing CMake presets on Windows with backslash

* adding checks to test

* Fix: parse -cc args when running "conan cache backup-upload"

* [fix] Add support for Git tree-less repository when getting the URL (#18444)

* Add test to validate Git treeless clone support

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Remove git method from the remote url

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* rsplit FTW

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Always return string

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* Filter out (fetch) and (push)

Signed-off-by: Uilian Ries <uilianries@gmail.com>

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>

* fix issue with missing folder in local-recipes-index (#18449)

* removed --local

* renamed editables->packages

* wip

* wip

* new test to check global.conf/-cc

---------

Signed-off-by: Uilian Ries <uilianries@gmail.com>
Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>
Co-authored-by: Uilian Ries <uilianries@gmail.com>
Co-authored-by: Carlos Zoido <mrgalleta@gmail.com>
Co-authored-by: LUCAS K DAL CASTEL <lucaskendrickdalcastel@gmail.com>

perseoGI committed:
Give wasm support to premake toolchain

franramirez688 committed:
Removed ObjC/C++ flags from C/C++ ones

franramirez688 committed:
wip

franramirez688 committed:
Added Objective C/C++ flags to both Gnu/Autotools toolchain

franramirez688 committed:
Fix

franramirez688 committed:
Fix

perseoGI committed:
Improved premake toolchain:

- Support fPIC option
- Support emcc compiler and other architectures
- Added premake exe and lib templates
- Improved test cases detecting new issues
- Fixed arch bug

perseoGI committed:
Added shared option support on premake libs

perseoGI committed:
Fix premake integration tests

rohel01 committed:
fix detection of riscv64 cpu in Meson toolchain

perseoGI committed:
Added architecture linker specific flags to toolchains

perseoGI committed:
Fix tests

memsharded committed:
implementing support for exe editables in CMakeConfigDeps (#18489)

* implementing support for exe editables in CMakeConfigDeps

* fix _package_framework var

perseoGI committed:
Changed dylib for so as test runs in linux

memsharded committed:
fix broken tests in windows

perseoGI committed:
Run premake test only in linux

perseoGI committed:
Remove print

Co-authored-by: Carlos Zoido <mrgalleta@gmail.com>

perseoGI committed:
Fixed elif issue in cmake blocks and added unit test

memsharded committed:
add fix to support git<2.36 (#18501)

AbrilRBS committed:
Fix issues with `self.generator_info` (#18503)

* Failing tests

* Make second one fail too

* Fixes

* Discard changes to conan/internal/model/conan_file.py

* Update test/integration/generators/test_generators_from_br.py

* Update test/integration/generators/test_generators_from_br.py

* Fix tests

perseoGI committed:
Update pyproject.toml

Co-authored-by: James <memsharded@gmail.com>

memsharded committed:
avoid leaking core.conf in user profiles (#18474)

* avoid leaking core.conf in user profiles

* Update conan/internal/model/conf.py

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

---------

Co-authored-by: Francisco Ramírez <franchuti688@gmail.com>

memsharded committed:
Improvement over ill-formed graphs with different visible=True/False for the same dependency (#18440)

* wip

* wip

* wip

* removed GraphRuntimeError in favor of GraphConflictError

* wip

* better removal of orphans

* simplified

* reduced warnings

* reducing warnings

* new output proposal

* new output warnings

ErniGH committed:
small fix sbom component bom-ref

franramirez688 committed:
[workspace] Changed `conanws.yml` format (#18493)

* Dynamically loading the name and version

* Fixed

* Fixed commands

* wip

* Changing tests

* Fixed tests

* Suggestions

* test

* wip

* Update conan/internal/model/workspace.py

Co-authored-by: James <memsharded@gmail.com>

* Update conan/api/subapi/workspace.py

Co-authored-by: James <memsharded@gmail.com>

* wip

* Raising error if duplicated references. Added test

---------

Co-authored-by: James <memsharded@gmail.com>

ErniGH committed:
1_6

perseoGI committed:
Added architecture linker specific flags to premake

ErniGH committed:
Add build_folder parameter in basic_layout (#18442)

* add build_folder

* test and use subproject path

perseoGI committed:
Added threads subsetting in emcc compiler model (#18520)

* Added threads subsetting on emcc compiler model

* Fixed test

* Return list instead

* Update test/integration/toolchains/meson/test_mesontoolchain.py

* Added integration tests to every toolchain

* WIP fix windows tests

* Fix windows autotools tests

---------

Co-authored-by: James <memsharded@gmail.com>

memsharded committed:
fix qcc cppstd support for latest QNX 8.0 for c++20 (#18538)

memsharded committed:
new 'conan cache ref <path>' to reverse-look the Conan cache (#18518)

* new 'conan cache ref <path>' to reverse-look the Conan cache

* fix

* fix bad paths stacktrace

* wip

* review

memsharded committed:
avoid incorrect  definitions in conf_info (#18541)

* avoid incorrect user.xxx definitions in conf_info

* test tools.xxx too

* more efficient checks

danimtb committed:
Fix: Fix cache check-intregrity output (#18544)

* Fix: Fix cache check-intregrity output

* bring indent back

* fix tests

* add missing package file test

* fix test

memsharded committed:
better error message for wrong cpp_info.requires (#18552)

franramirez688 committed:
Avoid library renames when Meson + Windows + MSVC static builds (#18533)

* wip

* Added splitext cmake function

* wip

* Added functional test

* Adding CMakeConfigDeps

* CMakeDeps backporting future change from CMake newer versions. Removed PkgConfigDeps change (too risky). Moved and modified test

* reducing diff

memsharded committed:
add assert message in CMakeConfigDeps (#18558)

memsharded committed:
better error message in CMakeConfigDeps for incorrect component requires (#18562)

memsharded committed:
fix incorrect inputs for relativize paths generator (#18561)

memsharded committed:
allow consuming meson libname.a libs in MSBuildDeps (#18557)

* allow consuming meson libname.a libs in MSBuildDeps

* fix test in OSX

czoido committed:
Avoid hyphens for msbuild verbosity argument passed to CMake after -- in powershell (#18548)

* replace - for /

* wip

* test powershell in integration

* simplify

* remove unused fixture

memsharded committed:
define CMake try-compile configuration (#18559)

* define CMake try-compile configuration

* risky fix

* try_compile only for known configurations

* add checks in tests

* guard cmake_try_compile_configuration

czoido committed:
Conan 2.19.0-dev

perseoGI committed:
Preamke generator: fix some issues with windows paths and MSBuild (#18572)

* Preamke generator: fix some issues with windows paths and MSBuild

* Added integration test case for msbuild_platform

AbrilRBS committed:
Revert Better remote caching  (#18586)

* Revert "better remote caching (#18411)"

This reverts commit ad76fa2cc44e5b5c9072475d062c29e593a12c17.

* Bump version

AbrilRBS committed:
Add gcc 12.5 (#18587)

memsharded committed:
Test refactor and improvements (#18576)

* improving tests

* rmeoved old config_path from tests

* wip

* wip

* wip

* wip

* fix tests

* removed xfailed test only fails in Windows if short-paths but not CI

AbrilRBS committed:
Fix `test_makefile_sysroot` test in Windows (#18595)

* Fix test_makefile_sysroot test in Windows

* Different paths for Windows

perseoGI committed:
Autotools: Added new makefile parameter to make function (#18578)

* Added new makefile parameter to make

* Use -f instead of --makefile option

* Use long option instead

Werni2A committed:
fix: Typo (#18571)

Co-authored-by: James <james@conan.io>

AbrilRBS committed:
Add freeze_support (#18603)

* Add freeze_support

* Freeze support call inside the __main__ method

franramirez688 committed:
[Part 2] Better remote caching (#18588)

* Let's avoid saving the whole object

* Caching only NotFoundException. Restoring remote.invalidate_cache() calls

* Cleaner

memsharded committed:
CMakeConfigDeps filter requires() to package_type=application (#18611)

perseoGI committed:
SSH runner: User friendly paramiko import protection (#18566)

* User friendly paramiko import protection

* Updated exception message

memsharded committed:
Test/improvements (#18616)

* modernizing tests to pytest

* more tests

* another batch

* finished refactor of unittest

* wip

* wip

AbrilRBS committed:
Debounce search input on html formats (#18575)

* Debounce search input

* Debounce search input for html graph too

* Focus from current pov

* Update conan/cli/formatters/report/diff_html.py

Co-authored-by: Ernesto de Gracia Herranz <ernestodegraciah@gmail.com>

---------

Co-authored-by: Ernesto de Gracia Herranz <ernestodegraciah@gmail.com>

memsharded committed:
enabling test_package_folder for conan export-pkg command (#18621)

memsharded committed:
Allow auto-detect with custom cc paths including spaces (#18628)

* wip

* wip

* fix tests

memsharded committed:
fix login messages hidden by verbosity level (#18642)

AbrilRBS committed:
Add explicit error when trying to export a reference with `channel` but no `user` (#18646)

* Error out if trying to export ref with channel and no user

* Add test

* PLay around with class attributes

* Fix error

* Remove redundant check

* Review

ErniGH committed:
add `@user/channel` to SBOM (#18649)

* add user channel to sbom

* Update conan/tools/sbom/cyclonedx.py

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

* fix

---------

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

AbrilRBS committed:
Improve version detection for cc compilers (#18600)

* Skip build information for gcc if present

* Simpler cc detection

AbrilRBS committed:
Let `conan source` reference the backup sources it generates in more cases (#18655)

* Let conan source reference the sources it downloads in more cases

* Test that the cache backup-upload also tries to upload non unknown references

perseoGI committed:
PremakeDeps: ensure correct linkage on dependent libraries (#18631)

* fix: 🐛 PremakeDeps was not providing correct link order to the linker

Co-authored-by: Abril Rincón Blanco <abrilrb@jfrog.com>

* Update conan/tools/premake/premakedeps.py

Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

* Fixed tests

* Removed redundant assert

* Improved test case on PremakeDeps

---------

Co-authored-by: Abril Rincón Blanco <abrilrb@jfrog.com>
Co-authored-by: Abril Rincón Blanco <5364255+AbrilRBS@users.noreply.github.com>

memsharded committed:
serialize conandata information (#18661)

AbrilRBS committed:
Add `cmake_target_aliases` support for `CMakeConfigDeps` (#18662)

* Add support for cmake_target_aliases

* Add aliases tests, coming from old cmakedeps

* Fix aliases for imported default target

* Remove redundant target store

* Restrict cmake version

* Typo

* Check for duplication caused by aliases

* Dont let users alias the global target

* Remove redundant test

AbrilRBS committed:
Create api helpers to help with initialization order and caching of common objects (#18644)

* Check conan version before loading hooks

* Make helpers class, name to be decided

* Rename, pep8 blank lines

* parametrize on reinit

* Reinit helpers

samuel-emrys commented:
Hi @memsharded this is looking great - is there anything blocking this from getting merged? I would find this useful.

memsharded commented:
No, nothing really blocking it besides the lack of time to move it forward.

We'd like to have some further feedback about the usage of these files by different tools, IDEs, and the like, to confirm these files are indeed useful/usable and convenient to consume.

samuel-emrys commented:
Thanks @memsharded - left some more detail about use case and feedback on the general approach here in https://github.com/conan-io/conan/issues/11042#issuecomment-3102634095

memsharded committed:
wip

memsharded committed:
wip

memsharded committed:
experimental

memsharded committed:
Merge branch 'develop2' into feature/dotenv

memsharded committed:
review

memsharded committed:
Merge branch 'develop2' into feature/dotenv

memsharded committed:
fix and new test
